integrations/datadog/client.py

```diff
@@ -0,0 +1,265 @@
+                url, params={"page[number]": page, "page[size]": page_size}
+            )
+
+            services = result.get("data")
+            if not services:
+                break
+
+            event.attributes.setdefault(CacheKeys.SERVICES, []).extend(services)
+            yield services
+            page += 1
+
+    async def get_slos(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+        """
+        Asynchronously fetches Datadog SLOs (Service Level Objectives).
+
+        This method retrieves SLOs from Datadog, handling pagination to ensure
+        all SLOs are fetched. If the SLOs are available in the cache, it retrieves
+        them from the cache.
+
+        Yields:
+            List[Dict[str, Any]]: A list of dictionaries representing Datadog SLOs.
+
+        Returns:
+            AsyncGenerator: An asynchronous generator yielding lists of SLOs.
+
+        Example:
+            async for slo_batch in your_instance.get_slos():
+                process_slo_batch(slo_batch)
+        """
+        if cache := event.attributes.get(CacheKeys.SLOS):
+            logger.info("Picking Datadog SLOs from cache")
+            yield cache
+            return
+
+        offset = 0
+        limit = MAX_PAGE_SIZE
+
+        while True:
+            url = f"{self.api_url}/api/v1/slo"
+            result = await self._send_api_request(
+                url, params={"limit": limit, "offset": offset}
+            )
+
+            slos = result.get("data")
+            if not slos:
+                break
+
+            event.attributes.setdefault(CacheKeys.SLOS, []).extend(slos)
+            yield slos
+            offset += limit
+
+    async def get_single_monitor(self, monitor_id: str) -> dict[str, Any] | None:
+        if not monitor_id:
+            return None
+        url = f"{self.api_url}/api/v1/monitor/{monitor_id}"
+        return await self._send_api_request(url)
+
+    async def create_webhooks_if_not_exists(self, app_host: Any, token: Any) -> None:
+        dd_webhook_url = (
+            f"{self.api_url}/api/v1/integration/webhooks/configuration/webhooks"
+        )
+
+        try:
+            webhook = await self._send_api_request(
+                url=f"{dd_webhook_url}/PORT", method="GET"
+            )
+            if webhook:
+                logger.info(f"Webhook already exists: {webhook}")
+                return
+        except httpx.HTTPStatusError as err:
+            if err.response.status_code == 404:
+                # Webhook does not exist, continue with creation
+                pass
+            elif err.response.status_code == 500:
+                # Webhooks are not yet enabled in Datadog
+                logger.error(err.response.text)
+            else:
+                raise
+
+        logger.info("Subscribing to Datadog webhooks...")
+
+        app_host_webhook_url = f"{app_host}/integration/webhook"
+        modified_url = insert_credentials(app_host_webhook_url, "port", token)
```

Comment:
you are passing user and token but the function calls them user and password, lets adjust it in the `insert_credentials` function.

Reply:
does it require the user to have `port` user at their datadog environment?

Reply:
Not necessarily, but that is the name I gave the user for the basic auth credentials.

[HTTP Basic Authentication](https://docs.datadoghq.com/integrations/webhooks/#http-basic-authentication)
If you want to post your webhooks to a service requiring authentication, you can use basic HTTP authentication by modifying your URL from `https://my.service.example.com` to `https://<USERNAME>:<PASSWORD>@my.service.example.com`

Reply:
lets add a comment about it